pkg/util/iptables/iptables.go

```diff
@@ -575,3 +576,235 @@ func IsNotFoundError(err error) bool {
 	}
 	return false
 }
+
+type RuleOption struct {
+	Arg     string
+	Negated bool
+}
+
+type Rule struct {
+	Modules []string
+	Jump    Chain
+	Comment string
+	Options map[string]RuleOption
+}
+
+type knownOption struct {
+	option      string
+	argRequired bool
+	canNegate   bool
+}
+
+func parseRule(line string, filterChains []Chain, filterChainPrefixes []string) (*Rule, Chain, error) {
+	knownOptions := []knownOption{
```

Comment:
Arrays can't be 'const' in Go, something about being able to take the address of an array element which would be illegal for a const.  It also shouldn't really be global since it's used in only one function, so I moved the struct declaration into the function too.